### ISSUE TITLE
Make GuiCelestialSelection extendeable

### ISSUE THREAD

**Issue by katzenpapst** [micdoodle8/Galacticraft#2085]
I just wanted to make my own version of GuiCelestialSelection, mostly to override teleportToSelectedBody(), and noticed that I have no choice than pretty much copypaste the entire class, since pretty much EVERYTHING is private. 

Now I've heard that it's considered to be "good practice" in java to make every goddamn thing private, but, is there an actual rational reason behind that? Would it be possible to change them to protected?

**Comment by radfast** [comment]
We don't really intend for GuiCelestialSelection to be over-ridden or extended by add-ons, therefore there is currently no API access to it (other than allowing Celestial Bodies to be registered of course).

If you are making a variant of it in your own mod, I don't see any problem with copy pasting the entire class.  You could argue that's actually **less** work for you, because you don't have to spend time thinking about which parts of it you want and which parts you don't want.

**Comment by radfast** [comment]
OK, I've actually changed my mind ... see 9878d4c :)

I think it's better for you not to copy + paste the whole class, in case we make future changes to GuiCelestialSelection.   The less in the class you copy / override, the more chance that our future changes will be effective even when your mod is installed alongside GC.

I'm sure there will be future changes, at minimum to add Planet/Moons descriptions, see #1557.

I have checked and apart from [GuiHandler](https://github.com/micdoodle8/Galacticraft/blob/ceb718af52f90e228dc914ce834a682e5d085d09/src/main/java/micdoodle8/mods/galacticraft/core/client/gui/GuiHandler.java) there's nothing else currently in Galacticraft that refers specifically to GuiCelestialSelection instead of a sub-class.   I'm not sure how you will handle GuiHandler ...
